InertiaCore/Utils/Vite.cs

```diff
@@ -0,0 +1,256 @@
+        builder.Attributes.Add("type", "text/javascript");
+        builder.Attributes.Add("src", asset(filePath));
+        return GetString(builder) + "\n\t";
+    }
+
+    // Generate a stylesheet tag for the given URL in HMR mode.
+    protected string makeStylesheetTag(string filePath)
+    {
+        var builder = new TagBuilder("link");
+        builder.Attributes.Add("rel", "stylesheet");
+        builder.Attributes.Add("href", asset(filePath));
+        return GetString(builder).Replace("></link>", " />") + "\n\t";
+    }
+
+    // Determine whether the given path is a CSS file.
+    protected bool isCssPath(string path)
+    {
+        return Regex.IsMatch(path, @".\.(css|less|sass|scss|styl|stylus|pcss|postcss)", RegexOptions.IgnoreCase);
+    }
+
+    // Generate React refresh runtime script.
+    public HtmlString reactRefresh()
+    {
+
+        if (!isRunningHot())
+        {
+            return new HtmlString("<!-- no hot -->");
+        }
+
+        var builder = new TagBuilder("script");
+        builder.Attributes.Add("type", "module");
+
+        var inner = String.Format(
+            "import RefreshRuntime from '{0}';", hotAsset("@react-refresh")) +
+            "RefreshRuntime.injectIntoGlobalHook(window);" +
+            "window.$RefreshReg$ = () => { };" +
+            "window.$RefreshSig$ = () => (type) => type;" +
+            "window.__vite_plugin_react_preamble_installed__ = true;";
+
+
+        builder.InnerHtml.AppendHtml(inner);
+
+        return new HtmlString(GetString(builder));
+    }
+
+    // Get the path to a given asset when running in HMR mode.
+    protected string hotAsset(string path)
+    {
+        var hotFilePath = getPublicDir(hotFile);
+        var hotContents = File.ReadAllText(hotFilePath);
+
+        return hotContents + "/" + path;
+
+    }
+
+    // Get the URL for an asset.
+    protected string asset(string path)
+    {
+        if (isRunningHot())
+        {
+            return hotAsset(path);
+        }
+
+        var pieces = new List<string>();
+        if (buildDirectory != null && buildDirectory != "")
+        {
+            pieces.Add(buildDirectory);
+        }
+        pieces.Add(path);
+        return "/" + String.Join("/", pieces);
+    }
+
+    protected bool isRunningHot()
+    {
+        return File.Exists(getPublicDir("hot"));
+    }
+
+    protected string GetString(IHtmlContent content)
+    {
+        var writer = new System.IO.StringWriter();
+        content.WriteTo(writer, HtmlEncoder.Default);
+        return writer.ToString();
+    }
+}
+
+public static class Vite
+{
+
+    // Set the filename for the manifest file.
+    public static string? useManifestFilename(string manifestFilename)
+    {
+        ViteBuilder.Instance.manifestFilename = manifestFilename;
+        return null;
+    }
+
+    // Set the Vite "hot" file path.
+    public static string? useHotFile(string hotFile)
+    {
+        ViteBuilder.Instance.hotFile = hotFile;
+        return null;
+    }
```

Comment:
Firstly, never return `null` when you don't want to return anything. Just use void:
```csharp
    public static void useHotFile(string hotFile)
```

However, we will return in these static methods the ViteBuilder instance, like this:
```suggestion
    // Set the Vite "hot" file path.
    public static IViteBuilder useHotFile(string hotFile)
    {
        ViteBuilder.Instance.hotFile = hotFile;
        return ViteBuilder.Instance;
    }
```
Using the interface, which I mentioned in the review description. This will allow us to stack them like this:
```csharp
Vite.useBuildDir(...)
    .usePublicDir(...)
    .useHotFile(...);
```

Reply:
So I implemented this locally, but not committed it yet. We need a terminal method, such as the input method, as this will return `InertiaCore.Utils.ViteBuilder` as a string the to the template when calling these functions. 

Our options are:
1. Document chaining/builder patterns as needing to end with `.input()`
2. Come up with a noop function that return void to finish the builder with. Maybe input also returns the instance and then `.build()` becomes the string result?
3. Change them to voids

Reply:
I went with option 1 for now. Usage:

```
Vite.useBuildDir(...)
    .usePublicDir(...)
    .useHotFile(...)
    .input(...);
```